### PR TITLE
Skip report callback

### PR DESCRIPTION
**Pull request by konraddysput**
# Why

This diff adds support for skip report callback. Users should use it to decide if the report should be filtered or not before we generate diagnostic information.

### CODE REVIEW

File: packages/sdk-core/tests/client/clientCallbacksTests.spec.ts [hunk 1]
```diff
@@ -47,4 +47,35 @@ describe('Client callbacks tests', () => {
...
+                skipReport: () => {
+                    return false;
+                },
+            });
+
+            await client.send(new Error());
+            expect(client.requestHandler.postError).toHaveBeenCalled();
+        });
+
+        it('Should send a report if the filter returns true ', async () => {
```
Comment (adamcronin42) [inline packages/sdk-core/tests/client/clientCallbacksTests.spec.ts:70]: 'Should _not_ send report..'

**Review by adamcronin42 — APPROVED** [review]
Looks good! Left one small comment on a typo.

File: packages/sdk-core/tests/client/clientCallbacksTests.spec.ts (hunk 1 above)
Comment (konraddysput) [reply packages/sdk-core/tests/client/clientCallbacksTests.spec.ts]: ups. Thank yoU!

**Comment by konraddysput** [comment]
> Do you think we could make the `skipReport` method asynchronous?

we cannot use asynchronous methods in the error reporting flow - every time we reach the asynchronous function, the unhandled exception handler won't wait on us. We can later also change the send method and not require it to be asynchronous. I will cover those details later.

**Pull request merged**